### DISCUSSION TITLE
Model Error during perfrom message_ix -> tutorial ->westeros_baseline #638

### DISCUSSION THREAD

**Discussion by Haoyran-Wuu** [iiasa/message_ix#713] [Getting started]
I got problems  when running the module "Time to Solve Model".
---------------------------------------------------------------------------
FileNotFoundError                         Traceback (most recent call last)
Cell In[29], line 1
----> 1 scenario.solve()

File [c:\Users\11456\anaconda3\envs\message_env\Lib\site-packages\message_ix\core.py:676](file:///C:/Users/11456/anaconda3/envs/message_env/Lib/site-packages/message_ix/core.py:676), in Scenario.solve(self, model, solve_options, **kwargs)
    656 def solve(self, model="MESSAGE", solve_options={}, **kwargs):
    657     """Solve MESSAGE or MESSAGE-MACRO for the Scenario.
    658 
    659     By default, :meth:`ixmp.Scenario.solve` is called with 'MESSAGE' as the
   (...)
    674         :class:`.GAMSModel`.
    675     """
--> 676     super().solve(model=model, solve_options=solve_options, **kwargs)

File [c:\Users\11456\anaconda3\envs\message_env\Lib\site-packages\ixmp\core\scenario.py:836](file:///C:/Users/11456/anaconda3/envs/message_env/Lib/site-packages/ixmp/core/scenario.py:836), in Scenario.solve(self, model, callback, cb_kwargs, **model_options)
    834 # Iterate until convergence
    835 while True:
--> 836     model_obj.run(self)
    838     # Store an iteration number to help the callback
    839     if not hasattr(self, "iteration"):
File [c:\Users\11456\anaconda3\envs\message_env\Lib\site-packages\message_ix\models.py:363](file:///C:/Users/11456/anaconda3/envs/message_env/Lib/site-packages/message_ix/models.py:363), in GAMSModel.run(self, scenario)
    361 optfile = Path(self.model_dir).joinpath("cplex.opt")
    362 lines = ("{} = {}".format(*kv) for kv in self.cplex_opts.items())
...
   1042 if "b" not in mode:
   1043     encoding = io.text_encoding(encoding)
-> 1044 return io.open(self, mode, buffering, encoding, errors, newline)

FileNotFoundError: [Errno 2] No such file or directory: 'C:GAMS\\for\\model\\files\\cplex.opt'

but I do have the file named "cplex.opt" in the directory, looking looking forward ur reply（begging~）

**Comment by glatterf42** [comment]
Hi Haoyran-Wuu, thank you for this question. Did you follow our [official installation instructions](https://docs.messageix.org/en/latest/install.html#gams-required), in particular step 3? Because it looks like your specific GAMS solver can't be found. Please make sure the `PATH` environment variable contains the path to your GAMS installation.
In addition, could you please provide the output of `message-ix show-versions` as run on the command line? This might further clarify the issue for us.

**Reply by Haoyran-Wuu** [reply]
Hi Glatter！ Thank you so much for your attention to my problem.  First of all, I have checked the configuration of GAMS path according to your suggestions, and the path has been added. Following, the output of "message-ix show-versions" is as follows:
ixmp:        3.7.0
message_ix:  3.6.1.dev83+g077c407.d20230523
     077c407 (HEAD -> main, origin/main, origin/HEAD) Retain meaning on typo fix
message_ix_models: None
message_data: None

click:       8.0.4
dask:        2022.7.0
genno:       installed
graphviz:    0.20.1
jpype:       1.4.0
… JVM path:  C:\Users\11456\anaconda3\Library\bin\server\jvm.dll
openpyxl:    3.0.10
pandas:      1.5.3
pint:        0.17
xarray:      2022.11.0
yaml:        6.0

iam_units:   installed
jupyter:     installed
matplotlib:  3.7.0
plotnine:    0.10.1
pyam:        1.3.1

GAMS:        29.1.0

python:      3.10.9 | packaged by Anaconda, Inc. | (main, Mar  1 2023, 18:18:15) [MSC v.1916 64 bit (AMD64)]
python-bits: 64
OS:          Windows
OS-release:  10
machine:     AMD64
processor:   Intel64 Family 6 Model 142 Stepping 10, GenuineIntel
byteorder:   little
LC_ALL:      None
LANG:        None
LOCALE:      ('Chinese (Simplified)_China', '936')

Look forward to your reply again, thank you!

**Reply by glatterf42** [reply]
Thanks for this information! Is there any particular reason for you to install version 3.7.0 of ixmp but not version 3.7.0 of message-ix? Both were released just last week, so if you started the installation process before that, you might still have an old version of message-ix. Could you try updating message-ix to version 3.7.0, please? 
However, following the installation instructions for anaconda, I don't think you should ever end up with a version like `3.6.1.dev83+g077c407.d20230523`, which looks more like you installed message-ix from a specific GitHub commit. So simply out of curiosity: did you mix the installation methods?

**Reply by Haoyran-Wuu** [reply]
Thank you very much for your answer. The answer to the question above is probably YES, because I reinstalled after the first exception was reported without a solution. I will try to update the version right away!

**Reply by glatterf42** [reply]
I see. Please let me know if the new version solves the issue.

**Reply by Haoyran-Wuu** [reply]
Hi Glatter, so sorry to bother u again. I synced message-ix and ixmp to 3.7.0. Running message-ix show-versions gave the following result
(message_env) C:\Users\11456>message-ix show-versions
ixmp:        3.7.0
message_ix:  3.7.0
message_ix_models: None
message_data: None

But the weird thing is, I ran the module and it gave the same output as I've mentioned before
---------------------------------------------------------------------------
FileNotFoundError                         Traceback (most recent call last)
Cell In[37], line 1
----> 1 scenario.solve()

File [~\AppData\Roaming\Python\Python311\site-packages\message_ix\core.py:676](https://file+.vscode-resource.vscode-cdn.net/c%3A/Users/11456/Desktop/%E8%A5%BF%E9%83%A8%E7%A2%B3%E4%B8%AD%E5%92%8C/%E7%A8%8B%E5%BA%8F/message_ix/tutorial/westeros/~/AppData/Roaming/Python/Python311/site-packages/message_ix/core.py:676), in Scenario.solve(self, model, solve_options, **kwargs)
    656 def solve(self, model="MESSAGE", solve_options={}, **kwargs):
    657     """Solve MESSAGE or MESSAGE-MACRO for the Scenario.
    658 
    659     By default, :meth:`ixmp.Scenario.solve` is called with 'MESSAGE' as the
   (...)
    674         :class:`.GAMSModel`.
    675     """
--> 676     super().solve(model=model, solve_options=solve_options, **kwargs)

File [~\AppData\Roaming\Python\Python311\site-packages\ixmp\core\scenario.py:836](https://file+.vscode-resource.vscode-cdn.net/c%3A/Users/11456/Desktop/%E8%A5%BF%E9%83%A8%E7%A2%B3%E4%B8%AD%E5%92%8C/%E7%A8%8B%E5%BA%8F/message_ix/tutorial/westeros/~/AppData/Roaming/Python/Python311/site-packages/ixmp/core/scenario.py:836), in Scenario.solve(self, model, callback, cb_kwargs, **model_options)
    834 # Iterate until convergence
    835 while True:
--> 836     model_obj.run(self)
    838     # Store an iteration number to help the callback
    839     if not hasattr(self, "iteration"):

File [~\AppData\Roaming\Python\Python311\site-packages\message_ix\models.py:363](https://file+.vscode-resource.vscode-cdn.net/c%3A/Users/11456/Desktop/%E8%A5%BF%E9%83%A8%E7%A2%B3%E4%B8%AD%E5%92%8C/%E7%A8%8B%E5%BA%8F/message_ix/tutorial/westeros/~/AppData/Roaming/Python/Python311/site-packages/message_ix/models.py:363), in GAMSModel.run(self, scenario)
    361 optfile = Path(self.model_dir).joinpath("cplex.opt")
    362 lines = ("{} = {}".format(*kv) for kv in self.cplex_opts.items())
...
   1042 if "b" not in mode:
   1043     encoding = io.text_encoding(encoding)
-> 1044 return io.open(self, mode, buffering, encoding, errors, newline)

FileNotFoundError: [Errno 2] No such file or directory: 'C:GAMS\\for\\model\\files\\cplex.opt'

So could you please tell me the reason why the operation fails again, and how to correct the solution?

**Reply by glatterf42** [reply]
I'm sorry that I don't have a lot of time to help you resolve this issue quickly. We are hosting the MESSAGEix Community Meeting this week, so a lot of time is spent on that. In the meantime, please make sure the cplex.opt file is in `C:GAMS\for\model\files\` for sure. Also note that you have to select the CPLEX solver upon installing GAMS to enable its usage later on. You might need a special license to make this solver work, but your error doesn't look like a typical 'invalid license' error to me. Please also try querying your favourite search engine to see if someone else encountered and solves your issue already. Again, sorry to provide only so generic help, but we are very busy this week.

**Reply by glatterf42** [reply]
Hi again, @Haoyran-Wuu. Were you able to resolve your problem? If so, could you please tell us what you did? If not, please let me know how I can help you :)
